connectivity/check/deployment.go

```diff
@@ -531,6 +532,16 @@ func (ct *ConnectivityTest) deploy(ctx context.Context) error {
 			if err != nil {
 				return err
 			}
+
+			svcHeadless := svc.DeepCopy()
+			svcHeadless.Name = echoOtherNodeDeploymentHeadlessServiceName
+			svcHeadless.Spec.ClusterIP = corev1.ClusterIPNone
+			svcHeadless.Spec.Type = corev1.ServiceTypeClusterIP
+			svcHeadless.ObjectMeta.Annotations["service.cilium.io/global-sync-endpoint-slices"] = "true"
+			_, err = ct.clients.src.CreateService(ctx, ct.params.TestNamespace, svcHeadless, metav1.CreateOptions{})
+			if err != nil {
+				return err
+			}
```

Comment:
Ideally, this should be guarded by the `GetService` check as above, to prevent failing if the service already exists. That said, this logic is already flawed in general, as we don't perform an upgrade in that case, hence it doesn't matter that much. Same for the one below.

Reply:
Indeed! I had to change the logic a bit for the copy to be a bit more convenient